imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java

```diff
@@ -509,8 +509,9 @@ private void clearAdLoadingInBackground() {
 
     @Override
     protected void onDestroy() {
-        log.d("IMA Start onDestroy");
-        boolean adManagerInitialized = (adsManager != null); // FEM-2600
+        log.d("IMA Start onDestroy lastAdEventReceived = " + lastAdEventReceived);
+        boolean adManagerInitialized = (adsManager != null || lastAdEventReceived != null); // FEM-2600
```

Comment:
Does this relate to the black screen issue? Why is this needed?

Reply:
Also, when is it possible that we don’t have adsManager but we have a last ad event revived from the ad? Is it possible?

Reply:
yes
no such option

Reply:
So why did you add this additional check?

Reply:
this is comming from another flow in app
on from destroy the activity but destroying the player

@OrenMe 
we have to give patch for this as well